ddp-workspace/projects/ddp-dsm-ui/src/app/utils/dynamic-value-util.model.ts

```diff
@@ -0,0 +1,47 @@
+export class DynamicValueUtilModel{
+
+  static camelCaseRegEx = new RegExp('(([a-z])+([A-z])+(.)*)*');
+  static convertToCamelCase(str: string): string {
+    if (!str)
+      {return str;}
+    str = String(str);  // Ensure str is treated as a string
+    const splittedWords = str.split('_');
+    if (splittedWords.length < 2) {
+      //in case there is no underscore in the column name, assume it is an older column name which is all uppercase
+      return this.handleAllUppercase(str);
+    }
+    return this.makeCamelCaseFrom(this.makeWordsLowerCase(splittedWords));
+  }
+
+  private static handleAllUppercase(word: string): string {
+    return this.isCamelCase(word) ? word : word.toLowerCase();
+  }
+
+  private static isCamelCase(word: string): boolean {
+    return this.camelCaseRegEx.exec(word).filter((value) => value === word).length > 0;
```

Comment:
@zyme this is the only change when moving the code to a separate class. I found out that using `RegExp.test()` checks if the reg expression is present in the string at all, not that if it matches the full string, so I am doing an exec, which finds all of the matches in the string, and then I check if the full string was one of the matches that were found